views/synonym-management.php

```diff
@@ -0,0 +1,167 @@
+								<option value="one-way" <?php echo $synonym['type'] === 'one-way' ? 'selected' : ''; ?>>
+									One-way
+								</option>
+								<option value="multi-way" <?php echo $synonym['type'] === 'multi-way' ? 'selected' : ''; ?>>
+									Multi-way
+								</option>
+							</select>
+						</div>
+						<div
+							class="blaze-wooless-synonym-row__input__key field-holder <?php echo $synonym['type'] === 'one-way' ? 'required' : 'hide'; ?>">
+							<label>Key <sup>*</sup></label>
+							<input class="blaze-wooless-synonym-row__input__key__input" type="text"
+								name="synonym[<?php echo $index; ?>][key]" value="<?php echo $synonym['key']; ?>"
+								placeholder="Enter key" <?php echo $synonym['type'] === 'one-way' ? 'required' : ''; ?>>
+						</div>
+						<div class="blaze-wooless-synonym-row__input__words field-holder required">
+							<label>Words <sup>*</sup></label>
+							<textarea class="blaze-wooless-synonym-row__input__words__input"
+								placeholder="Enter words separated by commas" name="synonym[<?php echo $index; ?>][words]"
+								required><?php echo implode( ', ', $synonym['words'] ); ?></textarea>
+						</div>
+						<div class="blaze-wooless-synonym-row__actions">
+							<button class="blaze-wooless-synonym-row__actions__remove">
+								<span class="dashicons dashicons-trash"></span>
+							</button>
+						</div>
+					</div>
+				</div>
+			<?php endforeach; ?>
+	</div>
+	<div id="blaze-wooless-synonym-add-row">
+		<button id="blaze-wooless-synonym-add-row__button button" type="button">Add Synonym</button>
+	</div>
+</div>
+
+<script id="blaze-wooless-synonym-template" type="text/x-jsrender">
+	<div class="blaze-wooless-synonym-row" data-row="{^{>row}}">
+		<div class="blaze-wooless-synonym-row__input">
+			<div class="blaze-wooless-synonym-row__input__type field-holder">
+				<label>Type</label>
+				<select class="blaze-wooless-synonym-row__input__type__select" name="synonym[{^{>row}}][type]">
+					<option value="one-way">One-way</option>
+					<option value="multi-way">Multi-way</option>
+				</select>
+			</div>
+			<div class="blaze-wooless-synonym-row__input__key field-holder required">
+				<label>Key <sup>*</sup></label>
+				<input class="blaze-wooless-synonym-row__input__key__input" type="text" name="synonym[{^{>row}}][key]" value="" placeholder="Enter key" required>
+			</div>
+			<div class="blaze-wooless-synonym-row__input__words field-holder required">
+				<label>Words <sup>*</sup></label>
+				<textarea class="blaze-wooless-synonym-row__input__words__input" placeholder="Enter words separated by commas" name="synonym[{^{>row}}][words]" required></textarea>
+			</div>
+			<div class="blaze-wooless-synonym-row__actions">
+				<button class="blaze-wooless-synonym-row__actions__remove">
+					<span class="dashicons dashicons-trash"></span>
+				</button>
+			</div>
+		</div>
+	</div>
+</script>
+<script type="text/javascript">
+	(function ($) {
```

Comment:
why not add this js into our main js file? is there a problem if we separate the script?

Reply:
i think we can keep it inline, since the script is only used on this page only